src/main/java/UI.java

```diff
@@ -0,0 +1,200 @@
+ */
+public class UI {
+
+    protected boolean isRunning;
+    Scanner sc = new Scanner(System.in);
+
+    /**
+     * Creates a UI object.
+     */
+    public UI() {
+        this.isRunning = true;
+    }
+
+    /**
+     * Prints out a welcome statement when users first open the Duke program.
+     */
+    public void welcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+
+        System.out.println("Hello I'm\n" + logo + "\nWhat can I do for you?");
+    }
+
+    /**
+     * Prints out a goodbye statement when the users enter bye, before the program ends.
+     */
+    public void goodbye() {
+
+        System.out.println("Bye. Hope to see you again soon!");
+        this.isRunning = false;
+
+    }
+
+    /**
+     * Prompts the user to enter into the program.
+     *
+     * @return the string that the user entered.
+     */
+    public String promptEntry() {
+        return sc.next();
+    }
+
+    /**
+     * Returns the formatted date and time. If it is in the wrong format, it will return the string entered.
+     *
+     * @param dateAndTime String value of the date and time.
+     * @return the formatted date and time, if it is in the correct format of dd/MM/yyyy hhmm
+     */
+    public static String getFormattedDate(String dateAndTime) {
+        String result = dateAndTime;
+        try {
+            Date date = new SimpleDateFormat("dd/MM/yyyy hhmm").parse(dateAndTime);
+            String day = new SimpleDateFormat("dd").format(date);
+            String month = new SimpleDateFormat("MMMMMMMMMMMMMMM").format(date);
+            String year = new SimpleDateFormat("yyyy").format(date);
+            String time = new SimpleDateFormat("h:mm a").format(date).toLowerCase();
+            String ordinalIndicator;
+
+            int int_day = Integer.parseInt(day);
+            if (int_day >= 11 && int_day <= 13) {
+                ordinalIndicator = "th";
+            } else if (int_day % 10 == 1) {
+                ordinalIndicator = "st";
+            } else if (int_day % 10 == 2) {
+                ordinalIndicator = "nd";
+            } else if (int_day % 10 == 3) {
+                ordinalIndicator = "rd";
+            } else {
+                ordinalIndicator = "th";
+            }
+
+            result = int_day + ordinalIndicator + " of " + month + " " + year + ", " + time;
+        } catch (ParseException e) {
+            System.out.println(e.getMessage());
+        }
+        return result;
+    }
+
+    /**
+     * Handles the commands entered by the user.
+     *
+     * @param command that the user entered.
+     * @param taskList the current tasklist, to be updated with every command.
+     */
+    public void handleCommand (String command, TaskList taskList) {
```

Comment:
In my opinion this method is a bit too lengthy. One way is resolving this is to use a separate Parser class instead.